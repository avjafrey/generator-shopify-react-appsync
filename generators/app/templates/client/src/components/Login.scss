.main {
    margin-top: 100px;
}

.header {
    text-align: center;
    margin-bottom: 50px;
}

.form {
    width: 400px;
    margin: auto;
}