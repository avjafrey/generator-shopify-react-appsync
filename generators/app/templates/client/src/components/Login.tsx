import { Button, DisplayText, FormLayout, TextField } from "@shopify/polaris";
import * as React from "react";

import { Spinner } from "./Spinner";

import * as styles from "./Login.scss";

interface ILoginProps {
    disableInstall: boolean;
    handleSubmit: (evt: React.FormEvent<HTMLFormElement>) => void;
    handleStoreChanged: (value: string, id: string) => void;
    errorMessage: string | undefined;
    installMessage: string;
    shop: string;
}

export function Login(props: ILoginProps) {
    return (
        <main role="main" className={styles.main}>
            <header className={styles.header}>
                <DisplayText size="extraLarge">Shopify App — Installation</DisplayText>
            </header>
            <div className={styles.form}>
                <form onSubmit={props.handleSubmit}>
                    <FormLayout>
                        <TextField
                            autoFocus={true}
                            error={props.errorMessage}
                            id="shop"
                            label="Please enter the “myshopify” domain of your store"
                            name="shop"
                            onChange={props.handleStoreChanged}
                            placeholder="example.myshopify.com"
                            value={props.shop}
                        />
                        {props.disableInstall && <Spinner />}
                        <Button
                            primary={true}
                            fullWidth={true}
                            size="large"
                            submit={true}
                            loading={props.disableInstall}
                        >
                            {props.installMessage}
                        </Button>
                    </FormLayout>
                </form>
            </div>
        </main>
    );
}
