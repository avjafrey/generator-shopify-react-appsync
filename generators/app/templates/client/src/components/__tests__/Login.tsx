import { AppProvider } from "@shopify/polaris";
import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import { Login } from "../Login";

const enptyFn = () => { return; };

test("Renders correctly with no error or shop", () => {
  const login = (
    <AppProvider>
      <Login
        disableInstall={false}
        handleStoreChanged={enptyFn}
        handleSubmit={enptyFn}
        errorMessage={undefined}
        installMessage="Install"
        shop={""}
      />
    </AppProvider>
  );

  const component = TestRenderer.create(login);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Renders correctly with shop", () => {
  const login = (
    <AppProvider>
      <Login
        disableInstall={false}
        handleStoreChanged={enptyFn}
        handleSubmit={enptyFn}
        errorMessage={undefined}
        installMessage="Install"
        shop={"example.myshopify.com"}
      />
    </AppProvider>
  );

  const component = TestRenderer.create(login);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Renders correctly with shop and disabled install", () => {
  const login = (
    <AppProvider>
      <Login
        disableInstall={true}
        handleStoreChanged={enptyFn}
        handleSubmit={enptyFn}
        errorMessage={undefined}
        installMessage="Install"
        shop={"example.myshopify.com"}
      />
    </AppProvider>
  );

  const component = TestRenderer.create(login);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test("Renders correctly with error", () => {
  const login = (
    <AppProvider>
      <Login
        disableInstall={false}
        handleStoreChanged={enptyFn}
        handleSubmit={enptyFn}
        errorMessage={"Store domain name is wrong"}
        installMessage="Install"
        shop={"example.myshopify.comx"}
      />
    </AppProvider>
  );

  const component = TestRenderer.create(login);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
