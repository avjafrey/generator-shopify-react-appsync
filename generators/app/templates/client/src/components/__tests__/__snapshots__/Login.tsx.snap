// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with error 1`] = `
<main
  role="main"
>
  <header>
    <p
      className="Polaris-DisplayText Polaris-DisplayText--sizeExtraLarge"
    >
      Shopify App — Installation
    </p>
  </header>
  <div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="Polaris-FormLayout"
      >
        <div
          className="Polaris-FormLayout__Item"
        >
          <div
            className=""
          >
            <div
              className="Polaris-Labelled__LabelWrapper"
            >
              <div
                className="Polaris-Label"
              >
                <label
                  className="Polaris-Label__Text"
                  htmlFor="shop"
                  id="shopLabel"
                >
                  Please enter the “myshopify” domain of your store
                </label>
              </div>
            </div>
            <div
              className="Polaris-TextField Polaris-TextField--hasValue Polaris-TextField--error"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              <input
                aria-describedby="shopError"
                aria-invalid={true}
                aria-labelledby="shopLabel"
                autoFocus={true}
                className="Polaris-TextField__Input"
                id="shop"
                name="shop"
                onChange={[Function]}
                placeholder="example.myshopify.com"
                style={null}
                value="example.myshopify.comx"
              />
              <div
                className="Polaris-TextField__Backdrop"
              />
            </div>
            <div
              className="Polaris-Labelled__Error"
              id="shopError"
            >
              <div
                className="Polaris-Labelled__ErrorIcon"
              >
                <span
                  className="Polaris-Icon"
                >
                  <svg
                    aria-hidden="true"
                    className="Polaris-Icon__Svg"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<path d=\\"M10 18a8 8 0 1 1 0-16 8 8 0 0 1 0 16zm-1-8h2V6H9v4zm0 4h2v-2H9v2z\\"  fill-rule=\\"evenodd\\"/>",
                      }
                    }
                    focusable="false"
                    viewBox="0 0 20 20"
                  />
                </span>
              </div>
              Store domain name is wrong
            </div>
          </div>
        </div>
        <div
          className="Polaris-FormLayout__Item"
        >
          <button
            className="Polaris-Button Polaris-Button--primary Polaris-Button--sizeLarge Polaris-Button--fullWidth"
            disabled={false}
            onMouseUp={[Function]}
            type="submit"
          >
            <span
              className="Polaris-Button__Content"
            >
              <span>
                Install
              </span>
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
</main>
`;

exports[`Renders correctly with no error or shop 1`] = `
<main
  role="main"
>
  <header>
    <p
      className="Polaris-DisplayText Polaris-DisplayText--sizeExtraLarge"
    >
      Shopify App — Installation
    </p>
  </header>
  <div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="Polaris-FormLayout"
      >
        <div
          className="Polaris-FormLayout__Item"
        >
          <div
            className=""
          >
            <div
              className="Polaris-Labelled__LabelWrapper"
            >
              <div
                className="Polaris-Label"
              >
                <label
                  className="Polaris-Label__Text"
                  htmlFor="shop"
                  id="shopLabel"
                >
                  Please enter the “myshopify” domain of your store
                </label>
              </div>
            </div>
            <div
              className="Polaris-TextField"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              <input
                aria-invalid={false}
                aria-labelledby="shopLabel"
                autoFocus={true}
                className="Polaris-TextField__Input"
                id="shop"
                name="shop"
                onChange={[Function]}
                placeholder="example.myshopify.com"
                style={null}
                value=""
              />
              <div
                className="Polaris-TextField__Backdrop"
              />
            </div>
          </div>
        </div>
        <div
          className="Polaris-FormLayout__Item"
        >
          <button
            className="Polaris-Button Polaris-Button--primary Polaris-Button--sizeLarge Polaris-Button--fullWidth"
            disabled={false}
            onMouseUp={[Function]}
            type="submit"
          >
            <span
              className="Polaris-Button__Content"
            >
              <span>
                Install
              </span>
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
</main>
`;

exports[`Renders correctly with shop 1`] = `
<main
  role="main"
>
  <header>
    <p
      className="Polaris-DisplayText Polaris-DisplayText--sizeExtraLarge"
    >
      Shopify App — Installation
    </p>
  </header>
  <div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="Polaris-FormLayout"
      >
        <div
          className="Polaris-FormLayout__Item"
        >
          <div
            className=""
          >
            <div
              className="Polaris-Labelled__LabelWrapper"
            >
              <div
                className="Polaris-Label"
              >
                <label
                  className="Polaris-Label__Text"
                  htmlFor="shop"
                  id="shopLabel"
                >
                  Please enter the “myshopify” domain of your store
                </label>
              </div>
            </div>
            <div
              className="Polaris-TextField Polaris-TextField--hasValue"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              <input
                aria-invalid={false}
                aria-labelledby="shopLabel"
                autoFocus={true}
                className="Polaris-TextField__Input"
                id="shop"
                name="shop"
                onChange={[Function]}
                placeholder="example.myshopify.com"
                style={null}
                value="example.myshopify.com"
              />
              <div
                className="Polaris-TextField__Backdrop"
              />
            </div>
          </div>
        </div>
        <div
          className="Polaris-FormLayout__Item"
        >
          <button
            className="Polaris-Button Polaris-Button--primary Polaris-Button--sizeLarge Polaris-Button--fullWidth"
            disabled={false}
            onMouseUp={[Function]}
            type="submit"
          >
            <span
              className="Polaris-Button__Content"
            >
              <span>
                Install
              </span>
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
</main>
`;

exports[`Renders correctly with shop and disabled install 1`] = `
<main
  role="main"
>
  <header>
    <p
      className="Polaris-DisplayText Polaris-DisplayText--sizeExtraLarge"
    >
      Shopify App — Installation
    </p>
  </header>
  <div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="Polaris-FormLayout"
      >
        <div
          className="Polaris-FormLayout__Item"
        >
          <div
            className=""
          >
            <div
              className="Polaris-Labelled__LabelWrapper"
            >
              <div
                className="Polaris-Label"
              >
                <label
                  className="Polaris-Label__Text"
                  htmlFor="shop"
                  id="shopLabel"
                >
                  Please enter the “myshopify” domain of your store
                </label>
              </div>
            </div>
            <div
              className="Polaris-TextField Polaris-TextField--hasValue"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              <input
                aria-invalid={false}
                aria-labelledby="shopLabel"
                autoFocus={true}
                className="Polaris-TextField__Input"
                id="shop"
                name="shop"
                onChange={[Function]}
                placeholder="example.myshopify.com"
                style={null}
                value="example.myshopify.com"
              />
              <div
                className="Polaris-TextField__Backdrop"
              />
            </div>
          </div>
        </div>
        <div
          className="Polaris-FormLayout__Item"
        >
          <div>
            <svg
              className="Polaris-Spinner Polaris-Spinner--colorTeal Polaris-Spinner--sizeLarge"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<path d=\\"M15.542 1.487A21.507 21.507 0 0 0 .5 22c0 11.874 9.626 21.5 21.5 21.5 9.847 0 18.364-6.675 20.809-16.072a1.5 1.5 0 0 0-2.904-.756C37.803 34.755 30.473 40.5 22 40.5 11.783 40.5 3.5 32.217 3.5 22c0-8.137 5.3-15.247 12.942-17.65a1.5 1.5 0 1 0-.9-2.863z\\" />",
                }
              }
              role="status"
              viewBox="0 0 44 44"
            />
          </div>
        </div>
        <div
          className="Polaris-FormLayout__Item"
        >
          <button
            aria-busy={true}
            className="Polaris-Button Polaris-Button--primary Polaris-Button--disabled Polaris-Button--loading Polaris-Button--sizeLarge Polaris-Button--fullWidth"
            disabled={true}
            onMouseUp={[Function]}
            role="alert"
            type="submit"
          >
            <span
              className="Polaris-Button__Content"
            >
              <span
                className="Polaris-Button__Spinner"
              >
                <svg
                  aria-label="Loading"
                  className="Polaris-Spinner Polaris-Spinner--colorWhite Polaris-Spinner--sizeSmall"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<path d=\\"M7.229 1.173a9.25 9.25 0 1 0 11.655 11.412 1.25 1.25 0 1 0-2.4-.698 6.75 6.75 0 1 1-8.506-8.329 1.25 1.25 0 1 0-.75-2.385z\\" />",
                    }
                  }
                  role="status"
                  viewBox="0 0 20 20"
                />
              </span>
              <span>
                Install
              </span>
            </span>
          </button>
        </div>
      </div>
    </form>
  </div>
</main>
`;
